File: app/app/src/main/java/com/cabal/app/Utils/Filters.java
package com.cabal.app.Utils;

import com.cabal.app.models.EventModel;

import java.util.ArrayList;
import java.util.List;

public class Filters {

    public static boolean AUTHOR_FILTER = false;

    public static boolean GUEST_FILTER = false;

    public static boolean FINISHED_FILTER = false;

    public static List<EventModel> performFiltering(List<EventModel> list) {
        //TODO: Upgrade after pumping Java to 8+
        List<EventModel> resultList = new ArrayList<>(list);
        if (AUTHOR_FILTER) {
            List<EventModel> modelsToRemove = new ArrayList<>();
            for (EventModel model : list) {
                if (!model.getCreator().equals(User.getUserNick())) {
                    modelsToRemove.add(model);
                }
            }
            resultList.removeAll(modelsToRemove);
        }
        if (GUEST_FILTER) {
            List<EventModel> modelsToRemove = new ArrayList<>();
            for (EventModel model : list) {
                if (model.getCreator().equals(User.getUserNick())) {
                    modelsToRemove.add(model);
                }
            }
            resultList.removeAll(modelsToRemove);
        }
        if (FINISHED_FILTER) {
            List<EventModel> modelsToRemove = new ArrayList<>();
            for (EventModel model : list) {
                if (!model.isFinished()) {
                    modelsToRemove.add(model);
                }
            }
            resultList.removeAll(modelsToRemove);
        }
        return resultList;
    }
}

File: app/app/src/main/java/com/cabal/app/Utils/Filters.kt
package com.cabal.app.Utils

import com.cabal.app.models.EventModel

object Filters {
        var AUTHOR_FILTER = false
        var GUEST_FILTER = false
        var FINISHED_FILTER = false

    fun performFiltering(list: List<EventModel>): List<EventModel> {
        val resultList = ArrayList<EventModel>(list)
        val modelsToRemove = ArrayList<EventModel>()
        if (AUTHOR_FILTER) list.forEach{ if (it.creator != User.getNick()) modelsToRemove.add(it) }
        if (GUEST_FILTER) list.forEach{ if (it.creator == User.getNick()) modelsToRemove.add(it) }
        if (FINISHED_FILTER) list.forEach{ if (it.isFinished) modelsToRemove.add(it) }
        resultList.removeAll(modelsToRemove)
        return resultList
    }
}